src/gwas.jl

```diff
@@ -298,6 +323,107 @@ function iht_path(
     return betas
 end
 
+"""
+    iht_path(x::BEDFile, y, path)
+
+If used with a `BEDFile` object `x`, then the temporary arrays are all initialized as `SharedArray`s of the proper dimensions.
+The additional optional arguments are:
+
+- `pids`, a vector of process IDs. Defaults to `procs(x)`.
+- `mask_n`, an `Int` vector used as a bitmask for crossvalidation purposes. Defaults to a vector of ones.
+"""
+function iht_path(
+    x        :: SnpLike{2},
+    y        :: Array{T,1},
+    path     :: DenseVector{Int};
+    pids     :: Vector{Int} = procs(x),
+    mask_n   :: Vector{Int} = ones(Int, size(y)),
+    tol      :: T    = convert(T, 1e-4),
+    max_iter :: Int  = 100,
+    max_step :: Int  = 50,
+    quiet    :: Bool = true
+) where {T <: Float}
+    print_with_color(:red, "gwas266, Starting iht_path().\n")
+    # size of problem?
+    n,p = size(x)
+
+    # how many models will we compute?
+    nmodels = length(path)
+
+    # also preallocate matrix to store betas
+    betas = spzeros(T,p,nmodels)  # a matrix to store calculated models
+
+    # preallocate temporary arrays
+    J = 1
+    v = IHTVariables(x, y, J, 1) # call Ben's code here, returns a different v
+
+    # compute the path
+    @inbounds for i = 1:nmodels
+
+        # model size?
+        q = path[i]
+
+        # monitor progress
+        quiet || print_with_color(:blue, "Computing model size $q.\n\n")
+
+        # these arrays change in size from iteration to iteration
+        # we must allocate them for every new model size
+        update_variables!(v, x, J*q)
+
+        # store projection of beta onto largest k nonzeroes in magnitude
+        project_k!(v.b, q)
```

Comment:
Instead of using `project_k!` you should use `project_group_sparse!` define in `MendelIHT_utilities.jl`. It is a new projection code that also takes group information into account.

Reply:
done